r2-core/src/main/java/com/linkedin/r2/transport/http/client/AsyncPoolImpl.java

```diff
@@ -255,13 +275,15 @@ public void start()
         throw new IllegalStateException(_poolName + " is " + _state);
       }
       _state = State.RUNNING;
-      if (_idleTimeout > 0)
+      long timeOut = Math.min(_idleTimeout, _waitTimeout);
+      if (timeOut > 0)
       {
-        long freq = Math.min(_idleTimeout / 10, 1000);
+        long freq = Math.min(timeOut / 10, 1000);
```

Comment:
waittimeout can be very low, let's add a min of 20-50 ms

Reply:
Sure,  20-50 ms will be too aggressive IMHO. But yea - this matters only when someone override the default.